src/commonMain/kotlin/io/github/detekt/sarif4k/JsonElementExtensions.kt

```diff
@@ -0,0 +1,64 @@
+package io.github.detekt.sarif4k
+
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+
+// Extracted from https://github.com/Kotlin/kotlinx.serialization/issues/296#issuecomment-1859572541
+private fun Any?.toJsonElement(): JsonElement = when (this) {
+    null -> JsonNull
+    is JsonElement -> this
+    is Map<*, *> -> toJsonElement()
+    is Collection<*> -> toJsonElement()
```

Comment:
Maybe, I don't know. I prefer to keep this function as it is and if we have problems in the future we can try to find solutions. This is something that `kotlinx.serialization` should provide.

Reply:
Yeah, I agree, but until that provides it, we are maintaining it. Iterable would cover a ton of extra things with no effort from us, because we just do `.map` anyway. It's changing a few words:
```suggestion
    is Iterable<*> -> toJsonElement()
```
and the extension function receiver below, PropertyBag.tags might need an adjustment.

Reply:
Done